bokehjs/src/lib/models/widgets/date_picker.ts

```diff
@@ -1,96 +1,76 @@
-import Pikaday from "pikaday"
+import flatpickr from "flatpickr"
 
 import {InputWidget, InputWidgetView} from "./input_widget"
 import {input} from "core/dom"
+import {CalendarPosition} from "core/enums"
 import * as p from "core/properties"
+import {isString, isArray} from "core/util/types"
 import {bk_input} from "styles/widgets/inputs"
-import "styles/widgets/pikaday"
-
-Pikaday.prototype.adjustPosition = function(this: Pikaday & {_o: Pikaday.PikadayOptions}): void {
-  if (this._o.container)
-    return
-
-  this.el.style.position = 'absolute'
-
-  const field = this._o.trigger!
-  const width = this.el.offsetWidth
-  const height = this.el.offsetHeight
-  const viewportWidth = window.innerWidth || document.documentElement!.clientWidth
-  const viewportHeight = window.innerHeight || document.documentElement!.clientHeight
-  const scrollTop = window.pageYOffset || document.body.scrollTop || document.documentElement!.scrollTop
-
-  const clientRect = field.getBoundingClientRect()
-  let left = clientRect.left + window.pageXOffset
-  let top = clientRect.bottom + window.pageYOffset
-
-  // adjust left/top origin to .bk-root
-  left -= this.el.parentElement!.offsetLeft
-  top -= this.el.parentElement!.offsetTop
-
-  // default position is bottom & left
-  if ((this._o.reposition && left + width > viewportWidth) ||
-      (this._o.position!.indexOf('right') > -1 && left - width + field.offsetWidth > 0))
-    left = left - width + field.offsetWidth
-
-  if ((this._o.reposition && top + height > viewportHeight + scrollTop) ||
-      (this._o.position!.indexOf('top') > -1 && top - height - field.offsetHeight > 0))
-    top = top - height - field.offsetHeight
-
-  this.el.style.left = left + 'px'
-  this.el.style.top = top + 'px'
+import "styles/widgets/flatpickr"
+
+type Date = string
+type DatesList = (Date | [Date, Date])[]
+
+function _convert_date_list(value: DatesList): any[] {
+  const result: any[] = []
+  for (const item of value) {
+    if (isString(item))
+      result.push(item)
+    else if (isArray(item) && item.length == 2)
+      result.push({from: item[0], to: item[1]})
+  }
+  return result
 }
 
 export class DatePickerView extends InputWidgetView {
   model: DatePicker
 
   protected input_el: HTMLInputElement
 
-  private _picker: Pikaday
+  private _picker: flatpickr.Instance
+
+  private _set(key: string, value: any): void {
+    if (this._picker != null) {
+      this._picker.set(key as any, value)
+    }
+  }
 
   connect_signals(): void {
     super.connect_signals()
-    this.connect(this.model.change, () => this.render())
+
+    const {value, min_date, max_date, disabled_dates, enabled_dates, position, inline} = this.model.properties
+    this.connect(value.change, () => this._set("defaultDate", value.value()))
```

Comment:
@bryevdv This sets only the initial date. The current date will not be changed. See the report at https://discourse.bokeh.org/t/datepicker-ui-not-updating-in-bokeh-2-0/5021

Reply:
@p-himik OK we should get a fix in for 2.0.1 (with a test) can you make an issue and I will take a look tomorrow.